### PR TITLE
Fix bugs of HTTPS support for opensds.

### PR DESCRIPTION
**Pull request by lasshpanzer**
**What this PR does / why we need it**:
Fix bugs of HTTPS support for opensds.

**Special notes for your reviewer**:
configuration steps for hotpot HTTPS access.
1.  export `OPENSDS_ENDPOINT`=https://localhost:50040, so client (osdsctl) HTTPS mode will be enabled. Client will import CA cert that is generated during the installation of the hotpot when the `HTTPS `scheme of the `OPENSDS_ENDPOINT` is detected. The CA cert will be used by client to verify the server is legal.
2. Modify the corresponding contents of the hotpot config file:
```
https_enabled=True 
beego_https_cert_file=/opt/opensds-security/opensds/opensds-cert.pem
beego_https_key_file=/opt/opensds-security/opensds/opensds-key.pem
```
then restart the server. Client can access server normally through HTTPS .
The HTTPS will be enabled and the opensds-cert.pem and opensds-key.pem genetrated during the installation will be imported when the server detects the flag `https_enabled` is true.
3. The hotpot dashboard currently does not support HTTPS.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func NewClient(c *Config) *Client {
 	case *KeystoneAuthOptions:
 		r = NewKeystoneReciver(c.AuthOptions.(*KeystoneAuthOptions))
 	default:
-		log.Printf("Warnning: Not support auth options, use default")
+		fmt.Println("Warnning: Not support auth options, use default")
```
Comment (stmcginnis) [inline client/client.go]: s/Warnning/Warning/

File: script/devsds/lib/opensds.sh [hunk 2]
```diff
@@ -21,15 +21,18 @@ set +o xtrace
...
-# is /opt/opensds-security/opensds/opensds-cert.pem, and key file is /opt/opensds-security/opensds/opensds-key.pem
+# Set global configuration.
 cat >> $OPENSDS_CONFIG_DIR/opensds.conf << OPENSDS_GLOBAL_CONFIG_DOC
 [osdslet]
 api_endpoint = 0.0.0.0:50040
 graceful = True
 log_file = /var/log/opensds/osdslet.log
 socket_order = inc
 auth_strategy = $OPENSDS_AUTH_STRATEGY
+#If https is enabled, the default value of cert file
```
Comment (stmcginnis) [inline script/devsds/lib/opensds.sh]: Should probably have a space after each # to be consistent.

Comment (lasshpanzer) [reply script/devsds/lib/opensds.sh]: Got it.

**Comment by leonwanghui** [comment]
One thing is could you apply these changes to `examples/opensds.conf`?

**Comment by leonwanghui** [comment]
And please also change `pkg/utils/testdata/opensds.conf`

File: pkg/utils/pwdTool/aes.go [hunk 3]
```diff
@@ -1,6 +1,6 @@
-// Copyright (c) 2019 Huawei Technologies Co., Ltd. All Rights Reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
+//// Copyright (c) 2019 Huawei Technologies Co., Ltd. All Rights Reserved.
+////
+//// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (leonwanghui) [inline pkg/utils/pwdTool/aes.go]: Please remove these lines

File: osdsctl/cli/pwd.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+func encrypter(cmd *cobra.Command, args []string) {
+	if len(args) == 0 {
+		cmd.Usage()
+		os.Exit(0)
+	}
+	ArgsNumCheck(cmd, args, 1)
+	// Encrypt the password
+	pwdTool := pwd.NewPwdTool("aes")
+	pwd, err := pwdTool.Encrypter(args[0])
+	fmt.Print(pwd)
```
Comment (leonwanghui) [inline osdsctl/cli/pwd.go]: Please use `fmt.Println()`

**Pull request merged**